zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -1926,18 +1926,25 @@ public void asyncCreate(final String path, Object datat, final CreateMode mode,
   private void doAsyncCreate(final String path, final byte[] data, final CreateMode mode,
       final long startT, final ZkAsyncCallbacks.CreateCallbackHandler cb,
       final String expectedSessionId) {
-    retryUntilConnected(() -> {
-      getExpectedZookeeper(expectedSessionId)
-          .create(path, data, ZooDefs.Ids.OPEN_ACL_UNSAFE, mode, cb,
-              new ZkAsyncRetryCallContext(_asyncCallRetryThread, cb, _monitor, startT, 0, false) {
-                @Override
-                protected void doRetry() {
-                  doAsyncCreate(path, data, mode, System.currentTimeMillis(), cb,
-                      expectedSessionId);
-                }
-              });
-      return null;
-    });
+    try {
+      retryUntilConnected(() -> {
+        getExpectedZookeeper(expectedSessionId)
+            .create(path, data, ZooDefs.Ids.OPEN_ACL_UNSAFE, mode, cb,
+                new ZkAsyncRetryCallContext(_asyncCallRetryThread, cb, _monitor, startT, 0, false) {
+                  @Override
+                  protected void doRetry() {
+                    doAsyncCreate(path, data, mode, System.currentTimeMillis(), cb,
+                        expectedSessionId);
+                  }
+                });
+        return null;
+      });
+    } catch (RuntimeException e) {
```

Comment:
nit, RuntimeException  -> Throwable?

Reply:
I thought about it but I think RuntimeException is good enough. If the RetryUntilConnected is changed to throw a checked exception, all the APIs would have to change and throw the checked exception. I just don't think the behavior would change. Let's keep the RuntimeException as it indicates we know what we are doing.